infrastructure/src/main/java/org/corfudb/infrastructure/InvokeCheckpointingJvm.java

```diff
@@ -0,0 +1,83 @@
+package org.corfudb.infrastructure;
+
+import org.corfudb.runtime.exceptions.NetworkException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class InvokeCheckpointingJvm implements IInvokeCheckpointing {
+
+    private static final long CONN_RETRY_DELAY_MILLISEC = 500;
+    private static final int MAX_COMPACTION_RETRIES = 8;
+    private final ServerContext serverContext;
+    private volatile Process checkpointerProcess;
+    private Logger syslog;
+
+    public InvokeCheckpointingJvm(ServerContext serverContext) {
+        this.serverContext = serverContext;
+        syslog = LoggerFactory.getLogger("SYSLOG");
+    }
+
+    @Override
+    public void invokeCheckpointing() {
+        for (int i = 1; i <= MAX_COMPACTION_RETRIES; i++) {
+            try {
+                String compactorScriptPath = (String) serverContext.getCompactorScriptPath();
+                String compactorConfigPath = (String) serverContext.getCompactorConfig();
+
+                syslog.trace("Script path: {}, configPath: {}", compactorScriptPath, compactorConfigPath);
+                List<String> endpoint = Arrays.asList(serverContext.getLocalEndpoint().split(":"));
+                String hostName = endpoint.get(0);
+                String port = endpoint.get(1);
+
+                if (this.checkpointerProcess != null && this.checkpointerProcess.isAlive()) {
+                    this.checkpointerProcess.destroy();
+                }
+
+                ProcessBuilder pb = new ProcessBuilder(compactorScriptPath, "--hostname", hostName, "--port",
```

Comment:
I believe there should be a way to run this code without running an external application. Can we?

Reply:
So the idea is to invoke a separate process to do checkpointing. Invoking directly will call checkpointers from within CorfuServer which we don't want to for these reasons -
1. If during checkpointing there's any sort of crash, it will cause server to crash - might be catastrophic as checkpointer failure shouldn't affect the server directly. 
3. To checkpoint the tables we need to use the CorfuRuntime. The corfu server currently doesn't share any data structures with the runtime created - so we won't exactly have any advantage by running it within the server as anyway same amount of memory is going to be used for checkpointing either way.

Reply:
Why not just run the compactor as a  separate process? Why do we have half of the functionality in corfu and the other part is run as an external process? Let's run the compactor independently from Corfu server?

Reply:
Good point. But the reason of having this particular orchestrator code here is, since clients do the checkpointing, the server spawns this process only if there are still some tables that the clients don't CP.
So most of the times this jvm doesn't even need to be invoked as the clients take care of CP.

Reply:
But that approach makes your architecture vague, people will be struggling to understand the motivation.
And if we are talking about cloud native approach, it's unacceptable to run additional processes inside the docker container, that's very unstable. So, we want to have a separate container for compaction.
I personally don't see how this kind of optimization when a server runs compactor actually helps. It's completely ok to run a process that just check the status of the tables and then runs compaction if needed